### PR TITLE
Add thumbnails

### PR DESCRIPTION
**Pull request by bnaul**
Currently just serves files from local storage, but should be very easy to adapt to use S3. Would appreciate some feedback though on the general architecture / JavaScript style.

EDIT: After changing the data setup around a bit, the handling of S3 access is going to happen at data-insert time; the set of changes here can be data location-agnostic, so it should be good to go as-is.

### CODE REVIEW

File: skyportal/app_server.py [hunk 1]
```diff
@@ -45,13 +46,19 @@ def make_app(cfg, baselayer_handlers, baselayer_settings):
...
 
+
         # User-facing pages
         (r'/.*', MainPageHandler)  # Route all frontend pages, such as
                                    # `/source/g647ba`, through the main page.
                                    #
                                    # Refer to Main.jsx for routing info.
     ]
 
+    if os.path.exists(cfg['paths:thumbnails']):
```
Comment (bnaul) [inline skyportal/app_server.py]: TODO serve `/thumbnails/` via nginx instead

File: requirements.txt [hunk 2]
```diff
@@ -7,3 +7,4 @@ bokeh==0.12.9
 pytest-randomly
 factory-boy==2.8.1
 pytest-factoryboy==1.3.1
+cloudstorage
```
Comment (stefanv) [inline requirements.txt]: Is this used anywhere?

File: skyportal/app_server.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import os.path
```
Comment (stefanv) [inline skyportal/app_server.py]: Revert changes to this file.

File: skyportal/models.py [hunk 4]
```diff
@@ -170,3 +221,30 @@ def from_ascii(cls, filename, source_id, instrument_id, observed_at):
         return cls(wavelengths=data[:, 0], fluxes=data[:, 1],
                    source_id=source_id, instrument_id=instrument_id,
                    observed_at=observed_at)
+
+
+#def format_public_url(context):
```
Comment (stefanv) [inline skyportal/models.py]: What should happen to this?

File: skyportal/plot.py [hunk 5]
```diff
@@ -145,6 +145,7 @@ def _plot_to_json(plot):
     return docs_json, render_items, custom_model_js
 
 
+# TODO async
```
Comment (stefanv) [inline skyportal/plot.py]: What was our decision here?

File: skyportal/models.py (hunk 4 above)
Comment (bnaul) [reply skyportal/models.py]: Just leave it there for now, there's nowhere better for it at the moment.

File: skyportal/plot.py (hunk 5 above)
Comment (bnaul) [reply skyportal/plot.py]: Not related to this PR, just a reminder. I clarified the TODO text a bit.

**Pull request merged**